src/main/java/frc/robot/utils/MovingAverage.java

```diff
@@ -0,0 +1,114 @@
+package frc.robot.utils;
+
+import java.io.*;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * The moving average contains a list of velocity based on the distance, and should give an approximation of the velocity to apply.
+ * In order to the moving average object to approximate better, you need to pass more values.
+ * The object calculate calculate a straight line between the closest point, and retrieve the velocity (in the straight line between the records).
+ */
+public class MovingAverage {
+    private final String DELIMITER = ",";
+    private final Map<Double, Double> distanceVelocityMap; // <key - distance, value - velocity>, tree map is automatically sort the values by distance.
```

Comment:
```suggestion
    private final TreeMap<Double, Double> distanceVelocityMap; // <key - distance, value - velocity>, TreeMap iterates according to the "natural ordering" of the keys.
```
Use `TreeMap` so it's clear you shouldn't pass a regular `Map` here.

Reply:
There is no need for that. I don't care if someone pass a map, the constructor sort a regular map

Reply:
But it's confusing that you use a `Map` but in the documentation mention `TreeMap`.

Reply:
Confuse who? the person who use this object doesn't care how I do it, as long as this works

Reply:
Then why do you need documentation at all, if they "don't care"?

Reply:
for me, if you want, I would delete this comment

Reply:
At least change "tree map" to "TreeMap", because I had to do some digging to understand what you're talking about (as someone who was not familiar with `TreeMap`s).